### PR TITLE
Fixing exports of namespaces `oas30` and `oas31`

### LINKED ISSUE #98: 4.0: the module exports OAS 3.1 only
I'm a bit confused.
If possible, could you please clarify how to operate OAS 3.0, if the module exports DSL and the model of 3.1 only?
Should I`import from "openapi3-ts/dist/cjs/model/openapi30"` directly or what exactly is supposed?
Maybe it should be explained in the changelog or Readme.

https://github.com/metadevpro/openapi3-ts/blob/26e17404b0844ff437487e830608eeca07d50b41/src/dsl/index.ts#L1

https://github.com/metadevpro/openapi3-ts/blob/26e17404b0844ff437487e830608eeca07d50b41/src/model/index.ts#L1

https://github.com/metadevpro/openapi3-ts/blob/26e17404b0844ff437487e830608eeca07d50b41/package.json#L7

### PR DESCRIPTION
**Pull request by RobinTail**
Alternative solution to #103.

Closes #98 

Explainted here: https://github.com/metadevpro/openapi3-ts/pull/103#issuecomment-1489074420

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
-export * from './dsl/index';
-export * from './model/index';
+export * as oas30 from "./oas30";
+export * as oas31 from "./oas31";
+export { Server, ServerVariable } from "./model/server"
```
Comment (RobinTail) [inline src/index.ts:3]: we can also have export of specification extension in the root if needed.
currently they are inside each of `oas30` and `oas31` only

File: src/index.spec.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
-import { OpenApiBuilder, Server, ServerVariable } from '.';
+import { oas30, oas31, Server, ServerVariable } from '.';
 
 describe('Top barrel', () => {
-    it('OpenApiBuilder is exported', () => {
-        const sut = OpenApiBuilder.create();
+    it('OpenApiBuilder v. 3.0 is exported', () => {
+        const sut = oas30.OpenApiBuilder.create();
         expect(sut).not.toBeNull;
+        expect(sut.rootDoc.openapi).toBe("3.0.0");
```
Comment (RobinTail) [inline src/index.spec.ts:8]: also added these assetions to check the correctness of the exported builders

**Comment by RobinTail** [comment]
I've built it locally and checked the imports and the types.
It seems to be operational

image

I confirm that all my tests run well with this implementation, including the integration test against the ESM build.
@pjmolina

**Review by pjmolina — APPROVED** [review]
Looks very good to me! Merging!

**Comment by RobinTail** [comment]
Thank you

**Pull request merged**